### DISCUSSION TITLE
getSession doesn't return sessionToken

### DISCUSSION THREAD

**Discussion by roosnic1** [nextauthjs/next-auth#2859] [Help] [answered]
I implemented a custom adapter which works with an existing backend. Everything seems to work but if i call `getSession({req: ctx})` i'm only getting:
```
{
  user: { name: 'User Name', email: 'user@email.com' },
  expires:  '2021-10-06T18:03:37.362Z'
}
```
Is there a reason why i don't see the `sessionToken`. In the callback `session({user, token, session})` i also don't see the `sessionToken`.  JWT are disabled in the options. Is this by design or am i doing something wrong?

**Comment by balazsorban44** [comment, marked as answer]
why would you need that? it's not something exposed directly, and shouldn't be either.

**Reply by roosnic1** [reply]
It might be that I'm doing something wrong but in my case the `sessionToken` is the same as in the cookie saved by next-auth. So it's already accessible by the user in the browser. Why is it hen omitted  in the object returned by `getSession()`?

**Reply by balazsorban44** [reply]
It is saved in an HttpOnly, secure cookie, so you cannot read it from the browser. it is a unique identifier to look up the corresponding session in the database. You likely won't need to use it directly. Could you explain why do you want it in the first place?

**Reply by balazsorban44** [reply]
FYI, in the future, we might even hash that `sessionToken` to make it even harder for the frontend to sniff after the session in the database.

**Reply by roosnic1** [reply]
I have a backend with a GraphQL API. Through my custom adapter i create a session in that backend and use the generated token as `sessionToken` in next-auth. My next.js app then uses GraphQL ApolloClient to access the same backend through server side GraphQL calls and direct client GraphQL calls. Some of those calls need to be authenticated.

**Reply by balazsorban44** [reply]
I am no security expert so take these words with a grain of salt...
But what I have seen is that usually, you don't want to send around the session identifier in your app. A more usual flow is that your identity server creates an `access_token` (in most cases a JWT), signs it, and sends that to the user, which you can attach to API requests from your client. Those backend APIs should have a way to decode the token and verify its integrity.


Nevertheless, the `sessionToken` **is** readable from your Next.js backend, from `req.cookies`. So you could proxy your client-side requests through an API route and use the `sessionToken` there.

**Reply by roosnic1** [reply]
Thank you very much for your help. I'll exactly do that. Do you think this will still be a usable solution once next-auth starts hashing the `sessionToken`?

**Reply by balazsorban44** [reply]
I'll have to learn more about how we can do that correctly, but I'm guessing it will be possible to retrieve the original value server-side
